apps/src/aichat/views/assets/FilePreview.tsx

```diff
@@ -0,0 +1,101 @@
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import {WithTooltip} from '@code-dot-org/component-library/tooltip';
+import {StrongText} from '@code-dot-org/component-library/typography';
+import classNames from 'classnames';
+import React, {useEffect, useRef, useState} from 'react';
+
+import aichatI18n from '@cdo/apps/aichat/locale';
+
+import styles from './staged-files-preview.module.scss';
+
+const FilePreview: React.FC<{
+  type: 'pdf' | 'image';
+  filename: string;
+  url: string;
+  isUploading?: boolean;
+  onRemove?: () => void;
+}> = ({type, filename, url, isUploading, onRemove}) => {
+  const imageRef = useRef<HTMLImageElement>(null);
+  const [imageLoaded, setImageLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
```

Comment:
All this (ref, state, useEffect) seems like a good candidate for a future hook/reusable image component!

Reply:
Oh good idea -- I saw your hidden file input one recently and thought it was a really good example of how hooks can be useful